services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java

```diff
@@ -0,0 +1,264 @@
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.cloudfront;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.InvalidKeyException;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.time.Instant;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.services.cloudfront.cookie.CookiesForCannedPolicy;
+import software.amazon.awssdk.services.cloudfront.cookie.CookiesForCustomPolicy;
+import software.amazon.awssdk.services.cloudfront.internal.auth.Pem;
+import software.amazon.awssdk.services.cloudfront.internal.auth.Rsa;
+import software.amazon.awssdk.services.cloudfront.internal.cookie.DefaultCookiesForCannedPolicy;
+import software.amazon.awssdk.services.cloudfront.internal.cookie.DefaultCookiesForCustomPolicy;
+import software.amazon.awssdk.services.cloudfront.internal.url.DefaultSignedUrl;
+import software.amazon.awssdk.services.cloudfront.internal.utils.SigningUtils;
+import software.amazon.awssdk.services.cloudfront.model.CloudFrontSignerRequest;
+import software.amazon.awssdk.services.cloudfront.url.SignedUrl;
+import software.amazon.awssdk.utils.IoUtils;
+import software.amazon.awssdk.utils.StringUtils;
+
+/**
+ *
+ * Utilities for working with CloudFront distributions
+ * <p>
+ *     To securely serve private content by using CloudFront, you can require that users access your private content by using
+ *     special CloudFront signed URLs or signed cookies. You then develop your application either to create and distribute signed
+ *     URLs to authenticated users or to send Set-Cookie headers that set signed cookies for authenticated users.
+ * </p>
+ * <p>
+ *     Signed URLs take precedence over signed cookies. If you use both signed URLs and signed cookies to control access to the
+ *     same files and a viewer uses a signed URL to request a file, CloudFront determines whether to return the file to the
+ *     viewer based only on the signed URL.
+ * </p>
+ *
+ */
+@Immutable
+@ThreadSafe
+@SdkPublicApi
+public final class CloudFrontUtilities {
+
+    private static final String CLOUDFRONT_NET = "cloudfront.net";
+
+    private CloudFrontUtilities() {
+    }
```

Comment:
Are intentionally not following the utilities pattern from S3, where things aren't static and they create CloudFrontUtilities directly, or indirectly from a CloudFrontClient?

Reply:
Went with private constructor since there are no class fields (region, etc. not needed)
Currently just call the methods as such: CloudFrontUtilities.getSignedUrlWithCannedPolicy(request)
Open to changing

Reply:
My concern is that this differs from the way we do it elsewhere, and if we moved to doing it the normal way in the future, customers would get warnings when they call these methods (because you're invoking a static method as an instance method).

Reply:
adding create() method as discussed